### ISSUE TITLE
Generator works different in javascript and typescript

### ISSUE THREAD

**Issue by worudso** [Microsoft/TypeScript#21261]
**TypeScript Version:**  2.6.2

**Search Terms:** 
generator, for
**Code**

```ts
function* foo(){
  yield 1;
  yield 2;
}

for (let o of foo()) {
  console.log(o);
  // expected output: 1

  break; // closes iterator, triggers return
}
// this code is from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of
```

**Expected behavior:**
print 1 like when I ran the exact same code as js.
**Actual behavior:**
print nothing.
**Playground Link:**  
https://www.typescriptlang.org/play/#src=function*%20foo()%7B%0D%0A%20%20yield%201%3B%0D%0A%20%20yield%202%3B%0D%0A%7D%0D%0A%0D%0Afor%20(let%20o%20of%20foo())%20%7B%0D%0A%20%20console.log(o)%3B%0D%0A%20%20%2F%2F%20expected%20output%3A%201%0D%0A%0D%0A%20%20break%3B%20%2F%2F%20closes%20iterator%2C%20triggers%20return%0D%0A%7D%0D%0A
**Related Issues:**

**Comment by HerringtonDarkholme** [comment]
You need to turn on `"downlevelIteration": true` to get full support for iterables.

**Comment by andy-ms** [comment]
If you have `--target es5` trying to use an iterator in `for-of` without `--downLevelIteration` should cause a compile error.
(If you have `--target ES6` the behavior should of course be the same as in JS because the output code will use a real `for-of` loop.)
